catalog/imaging/imaging-postprocess-plugin-nitf/src/main/java/org/codice/alliance/plugin/nitf/NitfPostProcessPlugin.java

```diff
@@ -0,0 +1,556 @@
+    if (StringUtils.isNotBlank(rootFileName)
+        && StringUtils.isNotBlank(rootFileName.replaceAll("[^A-Za-z0-9]", ""))) {
+      String strippedFilename = rootFileName.replaceAll(INVALID_FILENAME_CHARACTER_REGEX, "");
+      return String.format(DERIVED_IMAGE_FILENAME_PATTERN, qualifier, strippedFilename, extension)
+          .toLowerCase();
+    }
+
+    return String.format("%s.%s", qualifier, JPG).toLowerCase();
+  }
+
+  private byte[] scaleImage(final BufferedImage bufferedImage, int width, int height)
+      throws IOException {
+    BufferedImage thumbnail =
+        Thumbnails.of(bufferedImage)
+            .size(width, height)
+            .outputFormat(JPG)
+            .imageType(BufferedImage.TYPE_3BYTE_BGR)
+            .asBufferedImage();
+
+    ByteArrayOutputStream outputStream = new ByteArrayOutputStream();
+    ImageIO.write(thumbnail, JPG, outputStream);
+    outputStream.flush();
+    byte[] thumbnailBytes = outputStream.toByteArray();
+    outputStream.close();
+    return thumbnailBytes;
+  }
+
+  private byte[] renderToJpeg2k(final BufferedImage bufferedImage) throws IOException {
+
+    BufferedImage imageToCompress = bufferedImage;
+
+    if (bufferedImage.getColorModel().getNumComponents() == ARGB_COMPONENT_COUNT) {
+
+      imageToCompress =
+          new BufferedImage(
+              bufferedImage.getWidth(), bufferedImage.getHeight(), BufferedImage.TYPE_3BYTE_BGR);
+
+      Graphics2D g = imageToCompress.createGraphics();
+
+      g.drawImage(bufferedImage, 0, 0, null);
+    }
+
+    ByteArrayOutputStream os = new ByteArrayOutputStream();
+
+    J2KImageWriter writer = new J2KImageWriter(new J2KImageWriterSpi());
+    J2KImageWriteParam writeParams = (J2KImageWriteParam) writer.getDefaultWriteParam();
+    writeParams.setLossless(false);
+    writeParams.setCompressionMode(ImageWriteParam.MODE_EXPLICIT);
+    writeParams.setCompressionType("JPEG2000");
+    writeParams.setCompressionQuality(0.0f);
+
+    ImageOutputStream ios = new MemoryCacheImageOutputStream(os);
+    writer.setOutput(ios);
+    writer.write(null, new IIOImage(imageToCompress, null, null), writeParams);
+    writer.dispose();
+    ios.close();
+
+    return os.toByteArray();
+  }
+
+  private void addDerivedResourceAttribute(Metacard metacard, ProcessResource processResource) {
+    Attribute attribute = metacard.getAttribute(Core.DERIVED_RESOURCE_URI);
+    if (attribute == null) {
+      attribute = new AttributeImpl(Core.DERIVED_RESOURCE_URI, processResource.getUri());
+    } else {
+      AttributeImpl newAttribute = new AttributeImpl(attribute);
+      newAttribute.addValue(processResource.getUri());
+      attribute = newAttribute;
+    }
+
+    metacard.setAttribute(attribute);
+  }
+
+  private int calculateOverviewHeight(BufferedImage image) {
+    final int width = image.getWidth();
+    final int height = image.getHeight();
+
+    if (width >= height) {
```

Comment:
❓I'm not sure why we're comparing the `width` with the `height` here. Unless I'm missing something, shouldn't we be comparing the `width` with `maxSideLength` instead of adjust the height proportionally? We actually may need to check both, `height` and `width` first to see if they both exceed `maxSizeLength` and then use the one that exceeds it the most to compute the proper ratio.

With the current code, if `maxSideLength = 20`, `width = 10` and `height = 6`, this condition will be `(10 >= 6) = true` and this method will return `round(6 * (20 / 10)) = 3`. I would expect the height to remain the same and be 6 in this case.